src/rosecommon/src/itemdb.cpp

```diff
@@ -0,0 +1,60 @@
+#include "itemdb.h"
+#include "connection.h"
+
+using namespace RoseCommon;
+
+ItemDatabase::ItemDatabase() {
+  for ( int index = 1; index < 15; ++index ) {
+    _database[index].reserve( 1000 ); // We shouldn't need more then 1000 entries off the bat.
+  }
+}
+
+ItemDatabase::~ItemDatabase() {
+  _database.clear();
+}
+
+ItemDef::ItemDef(): type( 0 ),
+                    subtype( 0 ),
+                    id( 0 ),
+                    buy_price( 0 ),
+                    sell_price( 0 ),
+                    weight( 0 ),
+                    atk( 0 ),
+                    def( 0 ),
+                    range( 0 ),
+                    slots( 0 ),
+                    level( 0 ),
+                    view_id( 0 ) { }
+
+void ItemDatabase::initialize() {
+
+
+  // Load the item db table
+  auto conn = Core::connectionPool.getConnection( Core::osirose );
+  Core::ItemDBTable itemdb{};
+  try {
+    ItemDef newItem;
+    for ( uint8_t itemType = 1; itemType < 15; ++itemType ) {
+      for ( const auto& row : conn( sqlpp::select( sqlpp::all_of( itemdb ) ).from( itemdb ).where( itemdb.type == itemType ) ) ) {
+        newItem.type = static_cast<uint8_t>(row.type);
```

Comment:
are you sure you need to cast it? I thought it'd implicitly be cast to the correct size

Reply:
I don't want to implicitly cast it. I'm also doing this to prevent compiler warnings.

Reply:
Ok